static/js/util/test_utils.js

```diff
@@ -21,14 +22,47 @@ export function findCourse(courseSelector: (course: Course, program: Program) =>
   throw "Unable to find course";
 }
 
-export const alterFirstRun = (course: Course, overrideObject: Object): Course => {
+export const alterFirstRun = (course: Course, overrideObject: Object): CourseRun => {
   course.runs[0] = Object.assign({}, course.runs[0], overrideObject);
+  return course.runs[0];
 };
 
 export function findAndCloneCourse(courseSelector: (course: Course, program: Program) => boolean): Course {
   return _.cloneDeep(findCourse(courseSelector));
 }
 
+export function generateCourseFromExisting(courseToClone: Course, desiredRuns: number, runToCopy: ?CourseRun) {
+  let course = _.cloneDeep(courseToClone);
+  let currentRunCount = course.runs.length;
+  if (currentRunCount < desiredRuns) {
+    let courseRun = currentRunCount === 0 ? runToCopy : course.runs[0];
+    if (!courseRun) {
+      throw new Error('Need a course run to copy.');
+    }
+    let runsNeeded = desiredRuns - currentRunCount;
+    let idMax = _.max(_.map(course.runs, run => run.id)) || 0;
+    let positionMax = _.max(_.map(course.runs, run => run.position)) || 0;
+    for (let i = 0; i < runsNeeded; i++) {
+      let newCourseRun = _.cloneDeep(courseRun);
+      positionMax++;
+      idMax++;
+      newCourseRun = Object.assign(newCourseRun, {
+        position: positionMax,
+        id: idMax,
+        course_id: `${newCourseRun.course_id}-new-${i}`
+      });
+      course.runs.push(newCourseRun);
+    }
+  } else if (currentRunCount > desiredRuns) {
+    course.runs = _.take(course.runs, desiredRuns);
+  }
+  course = Object.assign(course, {
```

Comment:
`Object.assign` assigns to the first parameter. If you wanted a shallow clone it should be something like `Object.assign({}, ...`. Can you either remove the return value or add the `{}` parameter?